staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -271,9 +274,20 @@ func (h *kmsv2PluginProbe) check(ctx context.Context) error {
 
 	h.lastResponse = &kmsPluginHealthzResponse{err: nil, received: time.Now()}
 	h.ttl = kmsPluginHealthzPositiveTTL
+	h.keyID.Store(&p.KeyID)
 	return nil
 }
 
+// getCurrentKeyID returns the latest keyID from the Status() method or err if keyID is empty
+func (h *kmsv2PluginProbe) getCurrentKeyID(ctx context.Context) (string, error) {
+	go h.check(ctx)
+	keyID := *h.keyID.Load()
```

Comment:
I believe you need to initialize this with an empty string in `kmsPrefixTransformer` to avoid `nil` pointer panics.